tst/com/amazon/corretto/crypto/provider/test/ServiceSelfTestMetaTest.java

```diff
@@ -157,4 +157,13 @@ public void whenMultipleThreadsTest_doesNotDeadlock() throws Throwable {
 
         assertEquals(SelfTestStatus.PASSED, ((SelfTestResult)sneakyInvoke(selfTest, "runTest")).getStatus());
     }
+
+    @Test
+    public void givenACCPCacheSelfTestResultsPropertySetToFalse_whenRunTests_ExpectReset() throws Throwable  {
+        reset();
+        System.setProperty("com.amazon.corretto.crypto.provider.cacheselftestresults", "false");
+        accp = new AmazonCorrettoCryptoProvider();
+        assertTrue(SelfTestStatus.FAILED != accp.runSelfTests());
+        assertTrue(SelfTestStatus.FAILED != accp.reRunSelfTests());
```

Comment:
looks like this test just asserts that the self tests pass, regardless of caching status.

not sure if it's worth the effort, but is there some way we can assert that the tests are actually being re-run? perhaps by doing a `sneakyInvoke` on the `accp` instance to somehow corrupt it in a way that would cause the re-run to fail?

Reply:
Done